monai/auto3dseg/data_analyzer.py

```diff
@@ -0,0 +1,198 @@
+    ):
+        if path.isfile(output_path):
+            warnings.warn(f"File {output_path} already exists and will be overwritten.")
+            logger.debug(f"{output_path} will be overwritten by a new datastat.")
+
+        self.datalist = datalist
+        self.dataroot = dataroot
+        self.output_path = output_path
+        self.do_ccp = do_ccp
+        self.device = device
+        self.worker = worker
+        self.image_key = image_key
+        self.label_key = label_key
+
+    @staticmethod
+    def _check_data_uniformity(keys: List[str], result: Dict):
+        """
+        Check data uniformity since DataAnalyzer provides no support to multi-modal images with different
+        affine matrices/spacings due to monai transforms.
+
+        Args:
+            keys: a list of string-type keys under image_stats dictionary.
+
+        Returns:
+            False if one of the selected key values is not constant across the dataset images.
+
+        """
+
+        constant_props = [result[DataStatsKeys.SUMMARY][DataStatsKeys.IMAGE_STATS][key] for key in keys]
+        for prop in constant_props:
+            if "stdev" in prop and np.any(prop["stdev"]):
+                return False
+
+        return True
+
+    def get_all_case_stats(self):
+
+        files, _ = datafold_read(datalist=self.datalist, basedir=self.dataroot, fold=-1)
+        ds = data.Dataset(data=files)
+        dataset = data.DataLoader(ds, batch_size=1, shuffle=False, num_workers=self.worker, collate_fn=no_collation)
+
+        summarizer = SegSummarizer(self.image_key, self.label_key, do_ccp=self.do_ccp)
+        keys = list(filter(None, [self.image_key, self.label_key]))
+        transform_list = [
+            LoadImaged(keys=keys),
+            EnsureChannelFirstd(keys=keys),  # this creates label to be (1,H,W,D)
+            ToDeviced(keys=keys, device=self.device, non_blocking=True),
+            Orientationd(keys=keys, axcodes="RAS"),
+            EnsureTyped(keys=keys, data_type="tensor"),
+            Lambdad(keys=self.label_key, func=lambda x: torch.argmax(x, dim=0, keepdim=True) if x.shape[0] > 1 else x)
+            if self.label_key
+            else None,
+            SqueezeDimd(keys=["label"], dim=0) if self.label_key else None,
+            summarizer,
+        ]
+
+        tranform = Compose(list(filter(None, transform_list)))
+
+        result = {str(DataStatsKeys.SUMMARY): {}, str(DataStatsKeys.BY_CASE): []}
+
+        if not has_tqdm:
+            warnings.warn("tqdm is not installed. not displaying the caching progress.")
+
+        for batch_data in tqdm(dataset) if has_tqdm else dataset:
+            d = tranform(batch_data[0])
+            stats_by_cases = {
+                str(DataStatsKeys.BY_CASE_IMAGE_PATH): d[str(DataStatsKeys.BY_CASE_IMAGE_PATH)],
+                str(DataStatsKeys.BY_CASE_LABEL_PATH): d[str(DataStatsKeys.BY_CASE_LABEL_PATH)],
+                str(DataStatsKeys.IMAGE_STATS): d[str(DataStatsKeys.IMAGE_STATS)],
```

Comment:
Using the DataStatsKeys and other StrEnums was my initial implementation. However, everything works fine until I need to export the dictionary as json/yaml using ConfigParser. Because the dictionary keys are objects instead of real strings, json/yaml dumps are not working properly.

Reply:
I looked into this one and we can add a yaml represented for `StrEnum`:

```py
from monai.utils import StrEnum
from monai.bundle import config_parser

def streum_representer(dumper, data):
    return dumper.represent_scalar("tag:yaml.org,2002:str", str(data.value))

config_parser.yaml.SafeDumper.add_multi_representer(StrEnum, streum_representer)


from monai.utils import TraceKeys
test_case = {TraceKeys.CLASS_NAME: ["test", "test"]}
config_parser.ConfigParser.export_config_file(test_case, "my_test_8.json", fmt="json")
config_parser.ConfigParser.export_config_file(test_case, "my_test_8.yaml", fmt="yaml")

```

Reply:
I will include this change to the PR. Thanks!

Reply:
Added in f138cdbffeebe75698caca01eaad4b92ce193c9e.